packages/core/strapi/lib/factories.js

```diff
@@ -6,10 +6,20 @@ const { createController } = require('./core-api/controller');
 const { createService } = require('./core-api/service');
 const { createRoutes } = require('./core-api/routes');
 
+// Content type is proxied to allow for dynamic content type updates
+const getContentTypeProxy = (strapi, uid) => {
+  return new Proxy(strapi.contentType(uid), {
```

Comment:
I'm not really familiar with the proxy, could you explain a bit what's going on? (also why do we need the get(target, prop))

Reply:
a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) can wrap any javascript object, and intercept and redefine operations for that object.

With this you can extend how a `get` access behave, and leave all the other operations as they were original.
Whereas if we only define an object getter:

```js
{
  get contentType() {
    return strapi.contentType(uid)
  }
}
```
we are only exposing the getter of this property, and nothing else.

the get(target, prop) is the interface the tthe proxy get exposes:
- `target` : the original object you are proxying
- `prop`: the property being accessed